Neo4j.Driver/Neo4j.Driver/Internal/Driver.cs

```diff
@@ -39,7 +39,7 @@ internal Driver(Uri uri, IConnectionProvider connectionProvider, IRetryLogic ret
             _retryLogic = retryLogic;
         }
 
-        public ISession Session(AccessMode defaultMode=AccessMode.Write, string bookmark = null)
+        public ISessionAsync Session(AccessMode defaultMode=AccessMode.Write, string bookmark = null)
```

Comment:
You're certainly right about the ```Dispose``` point. Dispose is the facilitator of ```using``` block but more importantly is also called by the garbage collector to ensure every related resource is also disposed.

It is not arguable that most of our API contracts are designed to be used inside ```using``` blocks but there may be some use cases where it will not be an option. For this, I believe that we have to introduce Close & CloseAsync methods to required interfaces and call those from inside ```Dispose```.

We can talk about the naming alternatives and the way to introduce these async stuff into the API.